integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,675 @@
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.Info().Msgf("sent CCIP message that will get re-orged, msg id: %x", reorgingMsgEvent.Message.Header.MessageId)
+	msgEvent := testhelpers.TestSendRequest(t, e.Env, state, noreorgSourceSelector, destChainSelector, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destChainSelector].Receiver.Address().Bytes(), 32),
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.Info().Msgf("sent CCIP message that will not get re-orged, msgSentEvent: %x", msgEvent.Message.Header.MessageId)
+
+	// Run reorg above finality depth
+	const reorgDepth = 50
+	l.Info().
+		Int("reorgDepth", reorgDepth).
+		Uint64("sourceChainSelector", reorgSourceSelector).
+		Msg("starting blockchain reorg on Simulated Geth chain")
+	err = reorgSourceClient.GethSetHead(reorgDepth)
+	require.NoError(t, err, "error starting blockchain reorg on Simulated Geth chain")
+
+	nodeAPIs := dockerEnv.GetCLClusterTestEnv().ClCluster.NodeAPIs()
+	nonBootstrapCount := len(nodeAPIs) - 1
+	l.Info().Msgf("waiting for %d non-bootstrap nodes to report finality violation on the logpoller", nonBootstrapCount)
+	require.Eventually(t, func() bool {
+		violatedResponses := make(map[string]struct{})
+		for _, node := range nodeAPIs {
+			// skip bootstrap nodes, they won't have any logpoller filters
+			p2pKeys, err := node.MustReadP2PKeys()
+			require.NoError(t, err)
+
+			l.Debug().Msgf("got p2pKeys from node API: %+v", p2pKeys)
+
+			require.GreaterOrEqual(t, len(p2pKeys.Data), 1)
+			if !slices.Contains(nonBootstrapP2PIDs, p2pKeys.Data[0].Attributes.PeerID) {
+				l.Info().Msgf("skipping bootstrap node w/ p2p id %s", p2pKeys.Data[0].Attributes.PeerID)
+				continue
+			}
+
+			resp, _, err := node.Health()
+			require.NoError(t, err)
+			for _, d := range resp.Data {
+				if d.Attributes.Name == reorgLogPollerService &&
+					d.Attributes.Output == "finality violated" &&
+					d.Attributes.Status == "failing" {
+					violatedResponses[p2pKeys.Data[0].Attributes.PeerID] = struct{}{}
+					break
+				}
+			}
+
+			if _, ok := violatedResponses[p2pKeys.Data[0].Attributes.PeerID]; ok {
+				l.Info().Msgf("node %s reported finality violation", p2pKeys.Data[0].Attributes.PeerID)
+			} else {
+				l.Info().Msgf("node %s did not report finality violation, log poller response: %+v",
+					p2pKeys.Data[0].Attributes.PeerID,
+					getLogPollerHealth(reorgLogPollerService, resp.Data),
+				)
+			}
+		}
+
+		l.Info().Msgf("%d nodes reported finality violation", len(violatedResponses))
+		return len(violatedResponses) == nonBootstrapCount
+	}, 3*time.Minute, 5*time.Second, "not all the nodes report finality violation")
+	l.Info().Msg("All nodes reported finality violation")
+
+	// expect the commit to still go through on the non-reorged source chain.
```

Comment:
are you validating that reorged message is not processed

Reply:
Not explicitly, though I can add an assertion for that.